bundles/org.openhab.binding.mercedesme/src/main/java/org/openhab/binding/mercedesme/internal/api/Authorization.java

```diff
@@ -0,0 +1,533 @@
+    private final Logger logger = LoggerFactory.getLogger(Authorization.class);
+
+    private AccessTokenRefreshListener listener;
+    private AccessTokenResponse token = Utils.INVALID_TOKEN;
+    private String identifier;
+
+    protected static final String CONTENT_TYPE_URL_ENCODED = "application/x-www-form-urlencoded";
+    protected static final String CONTENT_TYPE_JSON = "application/json";
+
+    protected LocaleProvider localeProvider;
+    protected AccountConfiguration config;
+    protected Storage<String> storage;
+    protected HttpClient httpClient;
+    protected String baseUrl;
+
+    public Authorization(AccessTokenRefreshListener atrl, HttpClient hc, AccountConfiguration ac, LocaleProvider l,
+            Storage<String> store) {
+        listener = atrl;
+        httpClient = hc;
+        config = ac;
+        identifier = config.email;
+        localeProvider = l;
+        storage = store;
+
+        baseUrl = Utils.getLoginServer(config.region);
+        // restore token from persistence if available
+        String storedToken = storage.get(identifier);
+        if (storedToken != null) {
+            token = Utils.INVALID_TOKEN;
+            try {
+                TokenResponse tokenResponseJson = Utils.GSON.fromJson(storedToken, TokenResponse.class);
+                token = decodeToken(tokenResponseJson);
+            } catch (JsonSyntaxException e) {
+                logger.warn("Stored token {} for {} not parsable: {}", storedToken, config.email, e.getMessage());
+            }
+            if (!authTokenIsValid()) {
+                storage.remove(identifier);
+                logger.trace("Invalid token for {}", config.email);
+            } else {
+                atrl.onAccessTokenResponse(token);
+            }
+        } else {
+            logger.trace("No token for {} stored, stay on invalid token", config.email);
+        }
+    }
+
+    protected synchronized String getToken() {
+        if (token.isExpired(Instant.now(), EXPIRATION_BUFFER)) {
+            if (authTokenIsValid()) {
+                refreshToken();
+            }
+        }
+        return token.getAccessToken();
+    }
+
+    private void refreshToken() {
+        logger.trace("RefreshToken");
+        try {
+            String url = Utils.getTokenUrl(config.region);
+            Request req = httpClient.POST(url);
+            req.header("X-Device-Id", UUID.randomUUID().toString());
+            req.header("X-Request-Id", UUID.randomUUID().toString());
+
+            String grantAttribute = "grant_type=refresh_token";
+            String refreshTokenAttribute = "refresh_token="
+                    + URLEncoder.encode(token.getRefreshToken(), StandardCharsets.UTF_8.toString());
+            String content = grantAttribute + "&" + refreshTokenAttribute;
+            req.header(HttpHeader.CONTENT_TYPE, "application/x-www-form-urlencoded");
+            req.content(new StringContentProvider(content));
```

Comment:
See comment further down.
```suggestion
            req.content(new StringContentProvider(content), CONTENT_TYPE_URL_ENCODED);
```